File: SDKs/amazonaws.com/comprehend/2017-11-27/typescript/src/sdk/models/shared/keyphrasesdetectionjobproperties.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import { InputDataConfig } from "./inputdataconfig";
import { JobStatusEnum } from "./jobstatusenum";
import { LanguageCodeEnum } from "./languagecodeenum";
import { OutputDataConfig } from "./outputdataconfig";
import { VpcConfig } from "./vpcconfig";


// KeyPhrasesDetectionJobProperties
/** 
 * Provides information about a key phrases detection job.
**/
export class KeyPhrasesDetectionJobProperties extends SpeakeasyBase {
  @Metadata({ data: "json, name=DataAccessRoleArn" })
  dataAccessRoleArn?: string;

  @Metadata({ data: "json, name=EndTime" })
  endTime?: Date;

  @Metadata({ data: "json, name=InputDataConfig" })
  inputDataConfig?: InputDataConfig;

  @Metadata({ data: "json, name=JobArn" })
  jobArn?: string;

  @Metadata({ data: "json, name=JobId" })
  jobId?: string;

  @Metadata({ data: "json, name=JobName" })
  jobName?: string;

  @Metadata({ data: "json, name=JobStatus" })
  jobStatus?: JobStatusEnum;

  @Metadata({ data: "json, name=LanguageCode" })
  languageCode?: LanguageCodeEnum;

  @Metadata({ data: "json, name=Message" })
  message?: string;

  @Metadata({ data: "json, name=OutputDataConfig" })
  outputDataConfig?: OutputDataConfig;

  @Metadata({ data: "json, name=SubmitTime" })
  submitTime?: Date;

  @Metadata({ data: "json, name=VolumeKmsKeyId" })
  volumeKmsKeyId?: string;

  @Metadata({ data: "json, name=VpcConfig" })
  vpcConfig?: VpcConfig;
}

File: SDKs/amazonaws.com/comprehend/2017-11-27/go/pkg/models/shared/keyphrasesdetectionjobproperties.go
package shared

import (
	"time"
)

// KeyPhrasesDetectionJobProperties
// Provides information about a key phrases detection job.
type KeyPhrasesDetectionJobProperties struct {
	DataAccessRoleArn *string           `json:"DataAccessRoleArn,omitempty"`
	EndTime           *time.Time        `json:"EndTime,omitempty"`
	InputDataConfig   *InputDataConfig  `json:"InputDataConfig,omitempty"`
	JobArn            *string           `json:"JobArn,omitempty"`
	JobID             *string           `json:"JobId,omitempty"`
	JobName           *string           `json:"JobName,omitempty"`
	JobStatus         *JobStatusEnum    `json:"JobStatus,omitempty"`
	LanguageCode      *LanguageCodeEnum `json:"LanguageCode,omitempty"`
	Message           *string           `json:"Message,omitempty"`
	OutputDataConfig  *OutputDataConfig `json:"OutputDataConfig,omitempty"`
	SubmitTime        *time.Time        `json:"SubmitTime,omitempty"`
	VolumeKmsKeyID    *string           `json:"VolumeKmsKeyId,omitempty"`
	VpcConfig         *VpcConfig        `json:"VpcConfig,omitempty"`
}
